### ISSUE TITLE
What should object and function types print in {{value}}

### ISSUE THREAD

**Issue by justinbmeyer** [canjs/can-stache#521]
## Current Behavior

> NOTE: the current behavior changes depending if the value read is part of an observable (passed to can.view.live) or not.

The logic is governed by: 

- https://github.com/canjs/can-stache/blob/master/src/mustache_core.js#L355 - which branches if logic is observable or not, or the value is `!= "object" and weirdly `== "string"`
- https://github.com/canjs/can-fragment/blob/master/can-fragment.js#L104 - rules for object types
- https://github.com/canjs/can-view-live/blob/master/lib/core.js#L175 - rules for non object types

Primitives
- `null` and `undefined` -> ""
- `"string"` -> "string"
- `100` -> "100"`
- `NaN` -> "NaN"
- `true` -> "true"
- `false` -> "false"
- Symbol -> throws an error: `Cannot convert a Symbol value to a string` (might need to `.toString()` this)

Object types:
- ["<p>"] -> inserts a fragment with a paragraph element
- HTMLElement -> Inserts the paragraph element
- ["<p>", HTMLElement, [ HTMLElement ] ] -> inserts 3 elements
- `{}` -> [object Object]
- DefineMap -> [object Object]
- `function(){}` -> `function () { [native code] }` if on an observable, `""` if not

**Comment by matthewp** [comment]
I favor toStringing for consistency.

**Comment by justinbmeyer** [comment]
@matthewp for everything?  What about the Array & element behavior?  

Personally, I'd like objects to be written out.  So if had `{foo:"bar"}` I'd see that JSON written out to the page.

**Comment by justinbmeyer** [comment]
Some test code:

```
truthy: {{truthy}}, falsey: {{falsey}}, NaN: {{nan}}, Object: {{obj}},
        function: {{otherFunc.func}}
        {{#with otherFunc}}
            a: {{func}}
        {{/with}},
        defineMap: {{this}}
```

```
truthy: {default(){ return true; }},
        falsey: {default(){ return false; }},
        nan: {default(){ return NaN; }},
        obj: {default(){ return {}; }},
        func: {default(){ return function(){}; }},
        otherFunc: {type: "any", default(){ return {func: function(){}}; }},
```

**Comment by matthewp** [comment]
Yeah, I think it should do it consistently. The log helper is for pretty printing.

**Comment by justinbmeyer** [comment]
@matthewp even an HTMLElement?  I don't think we should toString that.  It's so natural to insert it.  Also, I think component's will depend on this.
